File: day12.py
class Solution(object):
    def largestPerimeter(self, nums):
        i = 0
        nums = sorted(nums)[::-1]
        for i in range(len(nums) - 2):
            if nums[i] < nums[i + 1] + nums[i + 2]:
                return nums[i] + nums[i + 1] + nums[i + 2]
        return 0

File: Day12.java
class Solution {
    public int largestPerimeter(int[] nums) {
        Arrays.sort(nums); 
        
        for(int i=nums.length-1;i>=2;i--){
            
            if(nums[i] < nums[i-1]+nums[i-2])
                return nums[i]+nums[i-1]+nums[i-2]; // if sum of two sides is less then cuurent side then it is a valid triangle
        }
        return 0;
    }
}
